Column[map(..., ...)] doesn't has default value

When using array(map), the following error will occur:
```
%TESTC  1 v2
%TSTTREE2,com.timeplus.jdbc.type.NestedTypeTest,true,1,false,1,NestedTypeTest,,[engine:junit-jupiter]/[class:com.timeplus.jdbc.type.NestedTypeTest]
%TSTTREE3,testNestedArrayMapType(com.timeplus.jdbc.type.NestedTypeTest),false,1,false,2,testNestedArrayMapType(),,[engine:junit-jupiter]/[class:com.timeplus.jdbc.type.NestedTypeTest]/[method:testNestedArrayMapType()]
%TESTS  3,testNestedArrayMapType(com.timeplus.jdbc.type.NestedTypeTest)

%ERROR  3,testNestedArrayMapType(com.timeplus.jdbc.type.NestedTypeTest)
%TRACES 
com.timeplus.exception.TimeplusSQLException: Column[map(int32, low_cardinality(nullable(string)))] doesn't has default value
        at com.timeplus.misc.ExceptionUtil.rethrowSQLException(ExceptionUtil.java:70)
        at com.timeplus.jdbc.statement.TimeplusPreparedInsertStatement.initBlockIfPossible(TimeplusPreparedInsertStatement.java:216)
        at com.timeplus.jdbc.statement.TimeplusPreparedInsertStatement.<init>(TimeplusPreparedInsertStatement.java:121)
        at com.timeplus.jdbc.TimeplusConnection.prepareStatement(TimeplusConnection.java:170)
        at com.timeplus.jdbc.type.NestedTypeTest.lambda$1(NestedTypeTest.java:128)
        at com.timeplus.jdbc.AbstractITest.withStatement(AbstractITest.java:148)
        at com.timeplus.jdbc.AbstractITest.lambda$0(AbstractITest.java:153)
        at com.timeplus.jdbc.AbstractITest.withNewConnection(AbstractITest.java:136)
        at com.timeplus.jdbc.AbstractITest.withStatement(AbstractITest.java:153)
        at com.timeplus.jdbc.type.NestedTypeTest.testNestedArrayMapType(NestedTypeTest.java:121)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
        at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
        at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
        at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:84)
        at org.eclipse.jdt.internal.junit5.runner.JUnit5TestReference.run(JUnit5TestReference.java:98)
        at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
        at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
        at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:757)
        at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
        at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
Caused by: com.timeplus.exception.NoDefaultValueException: Column[map(int32, low_cardinality(nullable(string)))] doesn't has default value
        at com.timeplus.data.IDataType.defaultValue(IDataType.java:36)
        at com.timeplus.data.type.complex.DataTypeArray.<init>(DataTypeArray.java:57)
        at com.timeplus.data.type.complex.DataTypeArray.lambda$0(DataTypeArray.java:41)
        at com.timeplus.data.DataTypeFactory.get(DataTypeFactory.java:75)
        at com.timeplus.data.DataTypeFactory.get(DataTypeFactory.java:60)
        at com.timeplus.data.Block.readFrom(Block.java:46)
        at com.timeplus.protocol.DataResponse.readFrom(DataResponse.java:32)
        at com.timeplus.protocol.Response.readFrom(Response.java:33)
        at com.timeplus.client.NativeClient.receiveResponse(NativeClient.java:207)
        at com.timeplus.client.NativeClient.receiveSampleBlock(NativeClient.java:129)
        at com.timeplus.jdbc.TimeplusConnection.getSampleBlock(TimeplusConnection.java:275)
        at com.timeplus.jdbc.statement.TimeplusPreparedInsertStatement.lambda$0(TimeplusPreparedInsertStatement.java:217)
        at com.timeplus.misc.ExceptionUtil.rethrowSQLException(ExceptionUtil.java:64)
        ... 76 more

%TRACEE 

%TESTE  3,testNestedArrayMapType(com.timeplus.jdbc.type.NestedTypeTest)

%RUNTIME2328
```

--- a/timeplus-native-jdbc/src/main/java/com/timeplus/data/type/complex/DataTypeMap.java
+++ b/timeplus-native-jdbc/src/main/java/com/timeplus/data/type/complex/DataTypeMap.java
@@ -73,6 +73,11 @@ public class DataTypeMap implements IDataType<Object, Object> {
     }
 
     @Override
+    public Map<?, ?> defaultValue() {
+        return Map.of(getNestedTypes()[0].defaultValue(), getNestedTypes()[1].defaultValue());
+    }
+
+    @Override
     public int getPrecision() {
         return 0;
     }

--- a/timeplus-native-jdbc/src/test/java/com/timeplus/jdbc/type/NestedTypeTest.java
+++ b/timeplus-native-jdbc/src/test/java/com/timeplus/jdbc/type/NestedTypeTest.java
@@ -21,6 +21,8 @@ import com.timeplus.jdbc.TimeplusStruct;
 
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
+import java.util.HashMap;
+import java.util.Map;
 import java.sql.Array;
 
 
@@ -114,5 +116,127 @@ public class NestedTypeTest extends AbstractITest implements BytesHelper {
         });
     }
 
+    @Test
+    public void testNestedArrayMapType() throws Exception {
+        withStatement(statement -> {
+            statement.execute("DROP STREAM IF EXISTS nested_test");
+            statement.execute(
+                    "CREATE STREAM IF NOT EXISTS nested_test (value array(map(int32, low_cardinality(nullable(string)))), "
+                    +"value2 map(string, array(low_cardinality(nullable(int))))) Engine=Memory()");
+
+            Integer rowCnt = 300;
+            try (PreparedStatement pstmt = statement.getConnection().prepareStatement(
+                    "INSERT INTO nested_test (value, value2) values(?, ?);")) {
+                for (int i = 0; i < rowCnt; i++) {
+                    // array of map
+                    Map<Integer, String> map1 = new HashMap<>();
+                    map1.put(1, null);
+                    map1.put(2, "test");
+                    Map<Integer, String> map2 = new HashMap<>();
+                    map2.put(1, null);
+                    map2.put(2, null);
+                    pstmt.setObject(1, statement.getConnection().createArrayOf("map(int32, low_cardinality(nullable(string)))", new Object[]{map1, map2}));
+
+                    // map (string, array)
+                    Array array = statement.getConnection().createArrayOf("low_cardinality(nullable(int))", new Integer[]{1, 2, null});
+                    Map<String, Object> map3 = new HashMap<>();
+                    map3.put("array", array);
+                    pstmt.setObject(2, map3);
+
+                    pstmt.addBatch();
+                }
+                pstmt.executeBatch();
+            }
+
+            ResultSet rs = statement.executeQuery("SELECT * FROM nested_test;");
+           // System.out.println("rs: " + rs.getRow());
+            int size = 0;
+            while (rs.next()) {
+                size++; 
+                // array of map
+                Array array2 = rs.getArray(1);
+                Object[] arrayValue2 = (Object[]) array2.getArray();
+                assertEquals(arrayValue2.length, 2);
+                Map<Integer, String> map1 = (Map<Integer, String>) arrayValue2[0];
+                assertEquals(map1.get(1), null);
+                assertEquals(map1.get(2), "test");
+                Map<Integer, String> map2 = (Map<Integer, String>) arrayValue2[1];
+                assertEquals(map2.get(1), null);
+                assertEquals(map2.get(2), null);
+
+                // map (string, array)
+                Map<String, Object> map3 = (Map<String, Object>) rs.getObject(2);
+                Array array3 = (Array) map3.get("array");
+                Object[] arrayValue3 = (Object[]) array3.getArray();
+                assertEquals(arrayValue3.length, 3);
+                assertEquals(arrayValue3[0], 1);
+                assertEquals(arrayValue3[1], 2);
+                assertEquals(arrayValue3[2], null);
+
+            }
+            assertEquals(size, rowCnt);
+            statement.execute("DROP STREAM IF EXISTS nested_test");
+        });
+    }
+
+    @Test
+    public void testNestedTupleMapType() throws Exception {
+        withStatement(statement -> {
+            statement.execute("DROP STREAM IF EXISTS nested_test");
+            statement.execute(
+                    "CREATE STREAM IF NOT EXISTS nested_test (value tuple(map(int32, low_cardinality(nullable(string))), low_cardinality(nullable(string))), "
+                    +"value2 map(string, tuple(low_cardinality(nullable(int))))) Engine=Memory()");
+
+            Integer rowCnt = 300;
+            try (PreparedStatement pstmt = statement.getConnection().prepareStatement(
+                    "INSERT INTO nested_test (value, value2) values(?, ?);")) {
+                for (int i = 0; i < rowCnt; i++) {
+                    // tuple (map, string)
+                    Map<Integer, String> map1 = new HashMap<>();
+                    map1.put(1, null);
+                    map1.put(2, "test");
+                    Object[] tupleValue = new Object[]{map1, "test"};
+                    TimeplusStruct tuple = new TimeplusStruct("tuple", tupleValue);
+                    pstmt.setObject(1, tuple);
+
+                    // map (string, tuple)
+                    Object[] tupleValue2 = new Object[]{3};
+                    TimeplusStruct tuple2 = new TimeplusStruct("tuple", tupleValue2);
+                    Map<String, Object> map2 = new HashMap<>();
+                    map2.put("tuple", tuple2);
+                    pstmt.setObject(2, map2);
+
+                    pstmt.addBatch();
+                }
+                pstmt.executeBatch();
+            }
+
+            ResultSet rs = statement.executeQuery("SELECT * FROM nested_test;");
+           // System.out.println("rs: " + rs.getRow());
+            int size = 0;
+            while (rs.next()) {
+                size++; 
+                // tuple (map, string)
+                Object obj1 = rs.getObject(1);
+                TimeplusStruct tuple = (TimeplusStruct) obj1;
+                Object[] tupleValue = (Object[]) tuple.getAttributes();
+                Map<Integer, String> map1 = (Map<Integer, String>) tupleValue[0];
+                assertEquals(map1.get(1), null);
+                assertEquals(map1.get(2), "test");
+                assertEquals(tupleValue[1], "test");
+
+                // map (string, tuple)
+                Map<String, Object> map2 = (Map<String, Object>) rs.getObject(2);
+                Object obj2 = (Object) map2.get("tuple");
+                TimeplusStruct tuple2 = (TimeplusStruct) obj2;
+                Object[] tupleValue2 = (Object[]) tuple2.getAttributes();
+                assertEquals(tupleValue2.length, 1);
+                assertEquals(tupleValue2[0], 3);
+
+            }
+            assertEquals(size, rowCnt);
+            statement.execute("DROP STREAM IF EXISTS nested_test");
+        });
+    }
    
 }

